### PR TITLE
Extract preprocessor module

### PR DESCRIPTION
**Pull request by lpil**
A start at breaking down the parser module in order to make it easier to understand.

### CODE REVIEW

File: config/config.exs [hunk 1]
```diff
@@ -3,6 +3,10 @@ use Mix.Config
 config :dogma,
   rule_set: Dogma.RuleSet.All,
 
+  exclude: [
+    ~r(\Alib/slime/doctype.ex\z),
```
Comment (lpil) [inline config/config.exs:8]: Ignored due to long lines in the literal doctypes.

File: test/preprocessor_test.exs [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+    assert result == [
+      "ul",
+      "  li.first",
+      "    a href='/a' A link",
+      "  li",
+      "    a href='/b' B link",
+    ]
+  end
+
+  test "inline tags are expanded when whitespace after the :" do
```
Comment (lpil) [inline test/preprocessor_test.exs:46]: This was broken before.

**Comment by doomspork** [comment]
@lpil this is looking really good!  Sorry I've been MIA, just got back into town.

**Comment by lpil** [comment]
Happy to merge?

**Comment by doomspork** [comment]
Go for it! :+1:

**Comment by doomspork** [comment]
Thanks again for all your contributions @lpil.  You've made some big changes! :grinning:

**Comment by lpil** [comment]
:D Thanks!

**Pull request merged**